frappe/model/base_document.py

```diff
@@ -220,7 +222,7 @@ def delete_key(self, key):
 		if key in self.__dict__:
 			del self.__dict__[key]
 
-	def append(self, key, value=None):
+	def append(self, key, value=None) -> Self:
```

Comment:
`doc.append` doesn't technically return self. They both have same base class somewhere higher up in the inheritance chain but it returns child table row. 

Use generics type for this? 

```python

T = TypeVar("T", bound="Document")

def append(self, key: str, value: T | dict | None = None) -> T:
    ...

Reply:
`Self` implies exact same object being returned (for use in chaining)

Reply:
![image](https://github.com/frappe/frappe/assets/9079960/596c56e4-9268-48d1-b1dd-801f86ea20b0)